tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp

```diff
@@ -679,7 +679,19 @@ void handleRequestImpl(sourcekitd_object_t ReqObj, ResponseReceiver Rec) {
     Opts.SyntaxTransferMode = syntaxTransferModeFromUID(TransferModeUID);
     Opts.SyntacticOnly = SyntacticOnly;
 
-    return Rec(editorReplaceText(*Name, InputBuf.get(), Offset, Length, Opts));
+    bool responded = false;
+    // If the client needs any information, send empty response ASAP.
+    if (!Opts.EnableStructure && !Opts.EnableDiagnostics &&
+        !Opts.EnableSyntaxMap &&
+        Opts.SyntaxTransferMode == SyntaxTreeTransferMode::Off) {
+      Rec(ResponseBuilder().createResponse());
```

Comment:
We need to synchronize with the client about the source edits, we should not return before we have have applied the edit in the document. What needs to be skipped is the syntactic parsing.

Reply:
I suggested to @rintaro offline that we bail out immediately after `EditorDoc->replaceText` in `SwiftLangSupport::editorReplaceText`.